CLI: remove --yul flag (#15316)
ethereum/solidity: Changelog.md

Compiler Features:
 * Command Line Interface: Do not perform IR optimization when only unoptimized IR is requested.
 * Commandline Interface: Add ``--transient-storage-layout`` output.

>  * Commandline Interface: Remove ``--yul`` flag.

Reviewer: Is there are reason we're doing this now? It is *technically* a [breaking change](https://github.com/ethereum/solidity/blob/develop/docs/050-breaking-changes.rst?plain=1#L158-L159).

Author: As per discussion with @ekpyron it is a breaking change mainly on CLI level, not really, effectively on language level b/c `--yul` isn't really usable anyways (although of course you are right, it _is_ breaking). Doing this to simplify yul dialect handling, prospectively getting rid of types, and finally make the whole numerical id integration easier.

 * SMTChecker: Replace CVC4 as a possible BMC backend with cvc5.
 * Standard JSON Interface: Do not perform IR optimization when only unoptimized IR is requested.
 * Standard JSON Interface: Add ``transientStorageLayout`` output.

>  * Yul: Dropping typed Yul dialects which could only be accessed via ``--yul`` in the CLI.

Reviewer 2: ```suggestion
 * Yul: Drop the deprecated typed Yul dialect that was only accessible via ``--yul`` in the CLI.
```
Not that we were loudly deprecating it to the outside world, but still :-).
